hax/treemakers/CorrectedDoubleS1Scatter.py

```diff
@@ -0,0 +1,386 @@
+        ds_s1_b_n_distinct_channels = 0
+        for i, channel in enumerate(s1_b_hitChannels):
+            if channel not in s1_a_hitChannels: 
+                ds_s1_b_n_distinct_channels += 1             
+    
+        result['ds_s1_b_n_distinct_channels'] = ds_s1_b_n_distinct_channels
+        
+        result['ds_s1_dt'] = peaks[s1_b].center_time - peaks[s1_a].center_time
+        
+        
+    #### Correction Map
+    
+        # Check that the correct S2 map is loaded and change if not
+        self.xy_map, self.loaded_xy_map_name = self.load_map("s2_xy_map",
+                                                             self.xy_map,
+                                                             self.loaded_xy_map_name)
+
+        # Load the 2D FDC map
+        self.fdc_2d_map, self.loaded_2d_fdc_map_name = self.load_map("fdc_2d",
+                                                                     self.fdc_2d_map,
+                                                                     self.loaded_2d_fdc_map_name)
+
+        # Load the 3D data driven FDC map
+        self.fdc_3d_map, self.loaded_3d_fdc_map_name = self.load_map("fdc_3d",
+                                                                     self.fdc_3d_map,
+                                                                     self.loaded_3d_fdc_map_name)
+
+        # Load the LCE map for TPF 2D FDC
+        self.lce_map_tpf_fdc_2d, self.loaded_lce_map_tpf_fdc_2d_name = self.load_map("s1_lce_map_tpf_fdc_2d",
+                                                                                     self.lce_map_tpf_fdc_2d,
+                                                                                     self.loaded_lce_map_tpf_fdc_2d_name)
+
+        # Load the LCE map for NN 3D FDC
+        self.lce_map_nn_fdc_3d, self.loaded_lce_map_nn_fdc_3d_name = self.load_map("s1_lce_map_nn_fdc_3d",
+                                                                                   self.lce_map_nn_fdc_3d,
+                                                                                   self.loaded_lce_map_nn_fdc_3d_name)
+
+        # Need the observed ('uncorrected') position.
+        # pax Interaction positions are corrected so lookup the
+        # uncorrected positions in the ReconstructedPosition objects
+        for rp in peaks[s2_a].reconstructed_positions:
+            if rp.algorithm == 'PosRecNeuralNet':
+                result['int_a_x_nn'] = rp.x
+                result['int_a_y_nn'] = rp.y
+                result['int_a_r_nn'] = np.sqrt(rp.x ** 2 + rp.y ** 2)
+                
+        for rp in peaks[s2_b].reconstructed_positions:
+            if rp.algorithm == 'PosRecNeuralNet':
+                result['int_b_x_nn'] = rp.x
+                result['int_b_y_nn'] = rp.y
+                result['int_b_r_nn'] = np.sqrt(rp.x ** 2 + rp.y ** 2)
+
+
+        int_a_z = interactions[int_a].z - interactions[int_a].z_correction
```

Comment:
This is z_observed (uncorrected) right? Should it be marked int_a_z_observed?

Reply:
Yes you're right. Did it